traitsui/tests/editors/test_range_editor.py

```diff
@@ -79,5 +79,33 @@ def test_large_range_slider_editor_set_with_text(self):
     def test_log_range_slider_editor_set_with_text(self):
         return self.check_set_with_text(mode='logslider')
 
+    @requires_toolkit([ToolkitName.qt])
     def test_range_text_editor_set_with_text(self):
         return self.check_set_with_text(mode='text')
+    
+    @requires_toolkit([ToolkitName.wx])
+    def test_range_text_editor_set_with_text_wx(self):
+        # this test is seperate to avoid an invalid state after deleting
+        # contents of textbox
```

Comment:
I find the conditions and comments slightly confusing... In fact, I think this test can be run with Qt as well but we are not running it - why?

Perhaps the logic in `check_set_with_text` is testing too much, there is not enough granularity.

What do you think about splitting up `check_set_with_text` into two scenarios: (1) The textbox is modified to another valid value and (2) the textbox is modified to an invalid value.

Then we can run tests for these two scenarios and for all styles... except scenario (2) with wx and text-style. Then we should open an issue for the behaviour and refer to that issue why the test is not exercised for wx.

Reply:
So, the weird behavior with wx text style isn't really that it miss handles invalid entries.  The discrepancy is that it gets angry when the textbox is completely empty and the others don't.  So then the 1st scenario, in order to include the wx text stye we would need to do something like just type 0, so the value is 10 which is in the range.  Then for the invalid test, we could type whatever out of the range, but that doesn't capture the discrepancy.